src/main/java/org/example/springintro/services/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+        Order order = findOrderById(orderId);
+        orderMapper.updateOrderFromDto(orderUpdateDto, order);
+        Order updatedOrder = orderRepository.save(order);
+        return orderMapper.toDto(updatedOrder);
+    }
+
+    @Override
+    public List<OrderItemDto> getOrderItems(Long orderId) {
+        List<OrderItem> orderItems = orderItemRepository.findByOrderId(orderId);
+        if (orderItems.isEmpty()) {
+            throw new IllegalArgumentException("No order items found for order ID: " + orderId);
+        }
+        return orderItems.stream().map(orderMapper::toOrderItemDto).toList();
+    }
+
+    @Override
+    public OrderItemDto getOrderItemByOrderIdAndItemId(Long orderId, Long itemId) {
+        List<OrderItem> orderItems = orderItemRepository.findByOrderId(orderId);
+        OrderItem orderItem = orderItems.stream()
+                .filter(item -> item.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Order item not found"));
+        return orderMapper.toOrderItemDto(orderItem);
+    }
+
+    private ShoppingCart getAndValidateCart(Long userId) {
+        ShoppingCart cart = shoppingCartRepository.findByUserId(userId);
+        if (cart == null || cart.getCartItems().isEmpty()) {
+            throw new IllegalStateException("Shopping cart is empty or not found");
+        }
+        return cart;
+    }
+
+    private Order createOrderFromCart(ShoppingCart cart, OrderRequestDto requestDto) {
+        Order order = new Order();
+        order.setUser(cart.getUser());
+        Set<OrderItem> orderItems = cart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(
+                            cartItem.getBook()
+                                    .getPrice()
+                                    .multiply(BigDecimal.valueOf(cartItem.getQuantity())));
+                    orderItem.setOrder(order);
+                    return orderItem;
+                })
+                .collect(Collectors.toSet());
+        order.setOrderItems(orderItems);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setTotal(calculateTotal(orderItems));
+        order.setStatus(Status.PENDING);
+        return order;
+    }
+
+    private BigDecimal calculateTotal(Set<OrderItem> orderItems) {
+        return orderItems.stream()
+                .map(OrderItem::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private Order findOrderById(Long orderId) {
+        return orderRepository.findById(orderId)
+                .orElseThrow(() -> new IllegalArgumentException("Order not found"));
```

Comment:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Order not found"));
```

provide order id to message